tests/integration/examples/utils.py

```diff
@@ -111,24 +120,23 @@ def run_example(
 
     yield example, runs
 
+    # Cleanup registered pipelines so they don't cause trouble in future
+    # example runs
+    for pipeline in client.depaginate(list_method=client.list_pipelines):
+        if pipeline.id not in existing_pipeline_ids:
+            client.delete_pipeline(pipeline.id)
```

Comment:
This will end up doing bad things when running tests in parallel or when running multiple CI batches at the same time on the same environment. Why don't you just use the `runs` yielded at the previous step instead ? Not 100% fail proof, but better than hacking off all pipelines created in the interval.

Reply:
Lots of examples run multiple pipelines (quickstart, mlflow deployment) in which case going by the runs doesn't work. Actually I'd argue the functionality to fetch/wait for those runs has the same issue when running in parallel? The only real solution to that would be clean workspaces for all those tests?

Reply:
Alright, I agree that lots of examples that you added in this PR run multiple pipelines, and that you need a solution that accommodates that, but I'm still not ok with hacking off *all* of the pipelines that are registered during the time that an example is running.

Currently, it is possible to run all of the examples in parallel on a single machine (the `run_example` utility takes in a pipeline name and only looks at runs from that pipeline). Your strategy makes that impossible.

Given that you do need to run and monitor multiple pipelines in some of the examples, I would suggest the following alternatives:

1. find a way to run one pipeline at a time (example command line arguments)
2. change the `run_example` signature to allow for a list of pipeline names to be supplied, then just use those pipeline names for monitoring and cleanup. These should help narrow the search and keeping the deletion confined to the running example.